test/TestUtilities/Test.Utility/Signing/CertificateUtilities.cs

```diff
@@ -35,5 +38,29 @@ internal static string GenerateRandomId()
         {
             return Guid.NewGuid().ToString();
         }
+
+        public static X509Certificate2 GetCertificateWithPrivateKey(X509Certificate bcCertificate, AsymmetricCipherKeyPair keyPair)
+        {
+#if IS_DESKTOP
+            RSA privateKey = DotNetUtilities.ToRSA(keyPair.Private as RsaPrivateCrtKeyParameters);
+
+            var certificate = new X509Certificate2(bcCertificate.GetEncoded());
+
+            certificate.PrivateKey = privateKey;
+#else
+            RSAParameters rsaParameters = DotNetUtilities.ToRSAParameters(keyPair.Private as RsaPrivateCrtKeyParameters);
+
+            var rsaCsp = new RSACryptoServiceProvider();
+
+            rsaCsp.ImportParameters(rsaParameters);
+
+            var privateKey = rsaCsp;
+
+            var certificateTmp = new X509Certificate2(bcCertificate.GetEncoded());
```

Comment:
I think you need to dispose of this.
```C#
using (var certificateTmp = new X509Certificate2(bcCertificate.GetEncoded()))
{
    certificate = RSACertificateExtensions.CopyWithPrivateKey(certificateTmp, privateKey);
}
```
Declare first in the method:
```C#
X509Certificate2 certificate;
```

Reply:
Thanks! Fixed.